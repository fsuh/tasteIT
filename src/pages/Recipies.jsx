import React, {useState, useEffect} from 'react';
import axios from 'axios';
import Card from '../components/Card';
import SearchBar from '../components/SearchBar';
import styles from '../styles/Recipies.module.css';

const Recipies = () => {

  const [data, setData] =useState([])
  const [searchInput, setSearchInput] = useState('');
  const [isLoading, setIsLoading]=useState(false);
  // const [flagArr, setFlagArr] = useState()


  useEffect(()=>{
    setIsLoading(true)

    axios.get('http://localhost:3001/recipe')
    .then((res)=>{
      setData(res.data);
      setIsLoading(false)
    })
    .catch((err)=>console.log(err.message));

    
  },[])

  if(isLoading){
    return <p>Loading...</p>
  }

  const handleChange = e =>{
    setSearchInput(e.target.value)
    
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('submit')
  }

  const search = (data) =>{
    return data.filter(
      (item) =>
      item.name.toLowerCase().includes(searchInput)||
      item.origin.toLowerCase().includes(searchInput))
  }

  return (
    <div className={styles.recipes}>
      <SearchBar 
        submit={handleSubmit}
        onChange={handleChange}
        value={searchInput}
      />

    <div className={styles.results}>

    {search(data).map((item) =>(
      <Card key={item.id}
        image={item.image}
        name={item.name}
        country={item.origin}

      />


    ))}

    </div>
    </div>
  )
}

export default Recipies