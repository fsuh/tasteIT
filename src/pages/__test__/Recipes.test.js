import { render } from "@testing-library/react";
import Recipies from "../Recipies";

describe("Recipes", () => {
  test("renders correctly", () => {
    const { container } = render(<Recipies />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test("matches snapshot", () => {
    const { asFragment } = render(<Recipies />);
    expect(asFragment()).toMatchSnapshot();
  });
});
