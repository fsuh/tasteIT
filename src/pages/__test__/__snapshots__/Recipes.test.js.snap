// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`Recipes renders correctly 1`] = `
<p>
  Loading...
</p>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`renders correctly 1`] = `
<p>
  Loading...
</p>
`;
